driver-core/src/test/java/com/datastax/driver/core/QueryLoggerErrorsTest.java

```diff
@@ -174,14 +174,9 @@ public void should_log_exception_from_the_given_result(PrimingRequest.Result res
         } catch(Exception e) {
             if(e instanceof NoHostAvailableException) {
                 assertThat(expectedException).isEqualTo(NoHostAvailableException.class);
-                // ok
                 Throwable error = ((NoHostAvailableException)e).getErrors().get(hostAddress);
                 assertThat(error).isNotNull();
-                if(loggedException.getSuperclass() == DriverInternalError.class) {
```

Comment:
As discussed, the exceptions reported now are the very same that we decode from the response. I think that the code initially tried to avoid exposing those because they inherit from `DriverInternalError`, which, in its author's initial conception, is a synonym of "bug in the driver". But since the Query Logger and JAVA-720, exceptions like `ServerError` or `OverloadedError` are first-class citizens, so I guess we can also pass them to the retry policy. The major mistake is to have created such exceptions as subclasses of `DriverInternalError` - but this cannot be changed, at least not in 2.1.